backend/src/types/RequestParams.ts

```diff
@@ -1,3 +1,20 @@
-export default interface IdParam {
-    id: string;
+export interface RoleIdParam {
+    roleId: string;
+}
+
+export interface UserIdParam {
+    userId: string;
+}
+
+export interface TypedRequest<bodyType, paramsType> extends Express.Request {
+    body: bodyType;
+    params: paramsType;
+}
+
+export interface TypedRequestBody<Type> extends Express.Request {
+    body: Type;
+}
+
+export interface TypedRequestParams<Type> extends Express.Request {
+    params: Type;
 }
```

Comment:
These are a great idea! It might make sense to move them to the `UtilTypes` file though?